packages/Search/src/details/DTK_DetailsTmp.hpp

```diff
@@ -0,0 +1,690 @@
+                                Kokkos::View<int *, DeviceType> &offset,
+                                Kokkos::View<int *, DeviceType> &ranks );
+    static void sort_results( Kokkos::View<int *, DeviceType> query_ids,
+                              Kokkos::View<int *, DeviceType> results,
+                              Kokkos::View<int *, DeviceType> ranks );
+    static void sort_results( Kokkos::View<int *, DeviceType> query_ids,
+                              Kokkos::View<int *, DeviceType> results,
+                              Kokkos::View<int *, DeviceType> ranks,
+                              Kokkos::View<double *, DeviceType> distances );
+    static void count_results( int n_queries,
+                               Kokkos::View<int *, DeviceType> query_ids,
+                               Kokkos::View<int *, DeviceType> &offset );
+    template <typename T>
+    static void
+    send_across_network( std::shared_ptr<Tpetra::Distributor> distributor,
+                         Kokkos::View<T *, DeviceType> exports,
+                         Kokkos::View<T *, DeviceType> imports );
+
+    static double epsilon;
+};
+
+template <typename DeviceType>
+double Tmp<DeviceType>::epsilon = 1.0e-6;
+
+template <typename DeviceType>
+template <typename T>
+void Tmp<DeviceType>::send_across_network(
+    std::shared_ptr<Tpetra::Distributor> distributor,
+    Kokkos::View<T *, DeviceType> exports,
+    Kokkos::View<T *, DeviceType> imports )
+{
+    auto exports_host = Kokkos::create_mirror_view( exports );
+    Kokkos::deep_copy( exports_host, exports );
+    auto imports_host = Kokkos::create_mirror_view( imports );
+    distributor->doPostsAndWaits(
+        Teuchos::ArrayView<T const>( exports_host.data(),
+                                     exports_host.extent( 0 ) ),
+        1, Teuchos::ArrayView<T>( imports_host.data(),
+                                  imports_host.extent( 0 ) ) );
+    Kokkos::deep_copy( imports, imports_host );
+}
+
+template <typename DeviceType>
+template <typename Query>
+void Tmp<DeviceType>::make_plan( Point epsilon,
+                                 Kokkos::View<Query *, DeviceType> queries,
+                                 BVH<DeviceType> const &bvh,
+                                 Kokkos::View<int *, DeviceType> &indices,
+                                 Kokkos::View<int *, DeviceType> &offset )
+{
+    int const n_queries = queries.extent( 0 );
+    Kokkos::View<Details::Overlap *, DeviceType> overlap_queries( "noname",
+                                                                  n_queries );
+
+    Kokkos::parallel_for( REGION_NAME( "noname" ),
+                          Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
+                          KOKKOS_LAMBDA( int q ) {
+                              Point point = queries( q )._query_point;
+                              Box box( {
+                                  point[0] - epsilon[0], point[0] + epsilon[0],
```

Comment:
Is `epsilon` really a `Point` or is it a tuple and you use a `Point` to store the data?